src/apis/impl/DataOwnerApiImpl.ts

```diff
@@ -71,24 +73,39 @@ export class DataOwnerApiImpl implements DataOwnerApi {
     keyPair?: { publicKey: string; privateKey: string }
   ): Promise<{ publicKey: string; privateKey: string }> {
     const dataOwnerId = this.getDataOwnerIdOf(user)
-
-    const { publicKey, privateKey } = keyPair ?? (await this._generateKeyPair())
-    const jwk = {
-      publicKey: spkiToJwk(hex2ua(publicKey)),
-      privateKey: pkcs8ToJwk(hex2ua(privateKey)),
-    }
-    await this.cryptoApi.cacheKeyPair(jwk)
-    await this.keyStorage.storeKeyPair(`${dataOwnerId}.${publicKey.slice(-32)}`, jwk)
-
     const dataOwner = await this.cryptoApi.getDataOwner(dataOwnerId).catch((e) => {
       throw this.errorHandler.createErrorFromAny(e)
     })
+
     if (dataOwner == null) {
       throw this.errorHandler.createErrorWithMessage(
         `The current user is not a data owner. You must be either a patient, a device or a healthcare professional to call this method.`
       )
     }
 
+    let jwks: { privateKey: JsonWebKey; publicKey: JsonWebKey } | undefined
+    if (keyPair === undefined && !overwriteExistingKeys && dataOwner.dataOwner.publicKey !== undefined) {
```

Comment:
It should be more than this: 
The dataOwner may have multiple publicKeys, we need to grab all the ones that are availables in the keyStorage. 
You can find them: 
- In the DataOwner.publicKey (ok); 
- In the DataOwner.aesExchangeKeys (as the first keys in the map hierarchy);
Make sure to use a set in case the "main" publicKey would be in the aesExchangeKeys